components/dot_menu/dot_menu.tsx

```diff
@@ -441,6 +465,24 @@ export class DotMenuClass extends React.PureComponent<Props, State> {
                             onClick={this.handleAddReactionMenuItemActivated}
                         />
                     </ChannelPermissionGate>
+                    <Menu.ItemAction
+                        id={`follow_post_thread_${this.props.post.id}`}
+                        show={(
+                            this.props.isCollapsedThreadsEnabled &&
+                                (
+                                    this.props.location === Locations.RHS_ROOT ||
+                                    this.props.location === Locations.RHS_COMMENT
+                                )
+                        )}
```

Comment:
I would think that a benefit of the post menu option added in this PR is to be able to follow within the center channel. Otherwise you can click the "Follow thread" button that's already at the top of the thread view right?

Reply:
> I would think that a benefit of the post menu option added in this PR is to be able to follow within the center channel

There is a "follow" button in the center channel, since yesterday I think, in the footer of the post :wink:

> Otherwise you can click the "Follow thread" button that's already at the top of the thread view right?

Actually the "Follow thread" is in the global thread viewer, but not in the RHS thread viewer.
Maybe that's confusing? cc @matthewbirtch

Reply:
![image](https://user-images.githubusercontent.com/3829551/118261929-dd9f7b80-b4bc-11eb-8838-9e91e25d9d37.png)

Reply:
https://github.com/mattermost/mattermost-webapp/pull/8002 is almost in for the RHSThread follow button.

Reply:
There does appear to be a center channel option https://www.figma.com/file/JmNM3WmTKHTKU5Cf7uNwOj/MM-18107-Threads-Overhaul-v3?node-id=1751%3A0

> In the case where there are no replies, the actions menu“Follow/Unfollow thread” menu item reads “Follow/Unfollow Message”

@matthewbirtch

1. Could you confirm if Following menuitem should only be visible when there are no replies? Or should it always be visible?

2. Should Reply menuitem be removed? (when CRT:ON)

Reply:
> In CenterChannel:
> 
> 1. could you confirm if Following menuitem should only be visible when there are no replies? Or should it always be visible?

I think users should always have the option to follow a message from the menu even if there are no replies yet. I just suggested a language change between "Follow/Unfollow Thread" vs" Follow/Unfollow Message"

> 2. Should Reply menuitem be removed? (when CRT:ON)

I don't see a reason to remove the reply menu item—at least for now. Users are still used to having it there, so there may be a transitional period where having it there redundantly is ok.